### DISCUSSION TITLE
Feature: Have a way to block traders from matching

### DISCUSSION THREAD

**Discussion by alecov** [bisq-network/bisq#6910] [Ideas]
Bisq needs a way to allow traders to block certain other traders, that is, to disallow these traders to match against your maker order, preferably before the trade. This is mostly useful for fiat payments but could exist for crypto swaps as well.

There are many situations in which a trader would want to undo a match.

Some traders configure bank payment accounts not as individuals, but as companies. I do not know if this is allowed by the protocol, but people do this. This has tax and other consequences in certain cases, and traders should have the possibility of avoiding this kind of trade on this (or any other, really) ground.

Some individuals are also ill-connected publicly or have bad reputation and I believe one should be able to vet one's trading partners to avoid problems.

Also for crypto payments, one might want to pass on eg. wallets that have interacted with mixers, since this reduces the ability to interact with CEXes in the future.

Or sometimes some traders just misbehave, or behave annoyingly. Traders offering payment from different accounts, or in multiple installments during the same trade period, or whatever bothersome attitude which does not justify mediation immediately but just makes the whole experience worse or worrisome. So a per-individual block during maker order creation should also exist.

I am not proposing that personal info should be revealed beforehand -- this defeats anonymity completely --, but to perhaps introduce an additional step at which some payment info (eg. account owner's name only) from both partners are exchanged, and either trader is allowed to back off the trade without consequences. This could be a opt-in, per-order flag so that if a trader does not want to engage with this protocol step, nothing happens and the trade is carried out exactly as it is now.

Let me know what you think about this.

**Comment by pazza83** [comment]
Bisq already has this.

You can block someone's onion address.

This stops you from being able to take each others' offers.

To do this copy the onion address of the peer you want to block then go to:

Setting > Preferences 

and add their onion address the ignored peers list.

You can block multiple onion addresses by separating them with a comma:

`address.onion:port,address.onion:port,address.onion:port`

**Reply by alecov** [reply]
>preferably before the trade.

What you suggest only works after you had made one trade with the peer.

**Reply by pazza83** [reply]
Yes sorry, was too quick to reply.

> perhaps introduce an additional step at which some payment info (eg. account owner's name only) from both partners are exchanged, and either trader is allowed to back off the trade without consequences

I cannot see this working as people would use it as an excuse to cancel trades when price moves against them.

Better I think to add peers to ignore list after you have traded with them if you would not like to have future trades with them.

**Comment by alecov** [comment]
> I cannot see this working as people would use it as an excuse to cancel trades when price moves against them.

It's a possibility, but again, people can opt-in into this feature and voluntarily assume any risks associated with it. I can see it having less liquidity in general but still working for people interested in this kind of protection.

The discussion below also ties into this one:
https://github.com/bisq-network/bisq/discussions/6634

I find it very risky to trade fiat with any anonymous individual in an era where your bank account can be frozen if it ever touches money considered "tainted" by the government.

Another idea: perhaps we could devise some set of tags (for example, "company-account" for a trader using a company account) which are visible for all traders, and traders can add or remove tags to a counterparty's account after a trade has been made. Similar to a reputation system, but with qualifiers; legitimate tags would be more numerous on an account with many trades.

Regardless of what the solution will be, I believe we need a way to reduce counterparty risk in a pre-match setting, that is, _before_ the match is settled.
